code/game/objects/random/misc.dm

```diff
@@ -520,3 +520,51 @@
 		/obj/item/reagent_containers/food/condiment/rice = 1,
 		/obj/item/reagent_containers/food/condiment/cocoa = 1
 	)
+
+/obj/random/maintenance_junk_or_loot
+	name = "random maintenance junk or loot"
+	desc = "Spawns any of: junk, loot, trash, trash pile, locker with junk or loot, (or other things)."
+	icon_state = "maint_junk_loot"
+	spawn_nothing_percentage = 25
+	has_postspawn = TRUE
+	problist = list(
+		// spawn just one thing:
+		/obj/random/junk = 3,
+		/obj/random/dirt_75 = 1,
+		/obj/random/loot = 1,
+		/obj/structure/trash_pile = 1,
+		/obj/random/tool = 0.2,
+		/obj/random/tech_supply = 0.1,
+		/obj/structure/girder = 0.1,
+		/obj/random/canister/empty = 0.1,
+		/obj/random/canister/filled = 0.05,
+		/obj/structure/closet/crate/loot = 0.01,
+
+		// spawn a container and maybe fill it with more junk:
+		/obj/structure/table/rack = 1,
+		/obj/structure/table/steel = 1,
+		/obj/structure/closet/crate = 1,
+		/obj/structure/closet/crate/trashcart = 0.2,
+		/obj/structure/closet/crate/plastic = 0.1,
+	)
+
+	var/list/more_junk_problist = list(
+		/obj/random/junk = 2,
+		/obj/random/loot = 1,
+		/obj/random/tech_supply = 0.2,
+		/obj/random/tool = 0.2,
+	)
+
+/obj/random/maintenance_junk_or_loot/post_spawn(var/obj/spawned)
+	if(istype(spawned, /obj/structure/table))
+		if(prob(65))
+			var/obj/more_junk = pickweight(more_junk_problist)
+			new more_junk(spawned.loc)
+	else if(istype(spawned, /obj/structure/closet))
+		if(prob(90))
+			var/i_max = rand(1, 4)
+			for(var/i in 1 to i_max)
```

Comment:
https://github.com/Aurorastation/Aurora.3/blob/master/.github/guides/STYLE.md#no-useless-variables-in-procs

`With few exceptions (mainly for readability), do not declare variables in procs that do not have any use [...] Exception to this would be _ in an interative loop, which by convention indicates a variable that is not important / to be ignored`

```suggestion
			for(var/_ in 1 to rand(1, 4))
```